feat: Add adjustable text size setting to interface

davecrab opened the pull request:
# Pull Request Checklist

### Note to first-time contributors: Please open a discussion post in [Discussions](https://github.com/open-webui/open-webui/discussions) to discuss your idea/fix with the community before creating a pull request, and describe your changes before submitting a pull request.

This is to ensure large feature PRs are discussed with the community first, before starting work on it. If the community does not want this feature or it is not relevant for Open WebUI as a project, it can be identified in the discussion before working on the feature and submitting the PR.

**Before submitting, make sure you've checked the following:**

- [x] **Target branch:** Verify that the pull request targets the `dev` branch. **Not targeting the `dev` branch will lead to immediate closure of the PR.**
- [x] **Description:** Provide a concise description of the changes made in this pull request down below.
- [x] **Changelog:** Ensure a changelog entry following the format of [Keep a Changelog](https://keepachangelog.com/) is added at the bottom of the PR description.
- [x] **Documentation:** If necessary, update relevant documentation [Open WebUI Docs](https://github.com/open-webui/docs) like environment variables, the tutorials, or other documentation sources.
- [x] **Dependencies:** Are there any new dependencies? Have you updated the dependency versions in the documentation?
- [x] **Testing:** Perform manual tests to **verify the implemented fix/feature works as intended AND does not break any other functionality**. Take this as an opportunity to **make screenshots of the feature/fix and include it in the PR description**.
- [x] **Agentic AI Code:** Confirm this Pull Request is **not written by any AI Agent** or has at least **gone through additional human review AND manual testing**. If any AI Agent is the co-author of this PR, it may lead to immediate closure of the PR.
- [x] **Code review:** Have you performed a self-review of your code, addressing any coding standard issues and ensuring adherence to the project's coding standards?
- [x] **Title Prefix:** To clearly categorize this pull request, prefix the pull request title using one of the following:
  - **BREAKING CHANGE**: Significant changes that may affect compatibility
  - **build**: Changes that affect the build system or external dependencies
  - **ci**: Changes to our continuous integration processes or workflows
  - **chore**: Refactor, cleanup, or other non-functional code changes
  - **docs**: Documentation update or addition
  - **feat**: Introduces a new feature or enhancement to the codebase
  - **fix**: Bug fix or error correction
  - **i18n**: Internationalization or localization changes
  - **perf**: Performance improvement
  - **refactor**: Code restructuring for better maintainability, readability, or scalability
  - **style**: Changes that do not affect the meaning of the code (white space, formatting, missing semi-colons, etc.)
  - **test**: Adding missing tests or correcting existing tests
  - **WIP**: Work in progress, a temporary label for incomplete or ongoing work

# Changelog Entry

### Description

- The default text size it unreadable for those with visual impairments. This PR adds a slider to make it more accessible, controls text size throughout the app.

- Introduces a user-configurable text size (scale) setting, accessible via a slider in the interface settings. 
- Updates CSS and Sidebar chat item components to respect the new --app-text-scale variable, and persists the setting in the store. 
- Adds related i18n strings and ensures the text scale is applied globally and clamped to allowed values.

### Added

- Shared text scale helpers (src/lib/utils/text-scale.ts) to centralize constants, clamping, and DOM updates.
Updated the global settings store (src/lib/stores/index.ts) to reuse the shared helpers and keep the CSS variable update in one place.
- Refactored the interface settings panel (src/lib/components/chat/Settings/Interface.svelte) to import the shared helpers.
- Slider to control text size app wide


### Changed

- [List any changes, updates, refactorings, or optimizations]

### Deprecated

- [List any deprecated functionality or features that have been removed]

### Removed

- [List any removed features, files, or functionalities]

### Fixed

- [List any fixes, corrections, or bug fixes]

### Security

- [List any new or updated security-related changes, including vulnerability fixes]

### Breaking Changes

- **BREAKING CHANGE**: [List any breaking changes affecting compatibility or functionality]

---

### Additional Information

- [Insert any additional context, notes, or explanations for the changes]
  - [Reference any related issues, commits, or other relevant information]

### Screenshots or Videos

- [Attach any relevant screenshots or videos demonstrating the changes]

### Contributor License Agreement

By submitting this pull request, I confirm that I have read and fully agree to the [Contributor License Agreement (CLA)](https://github.com/open-webui/open-webui/blob/main/CONTRIBUTOR_LICENSE_AGREEMENT), and I am providing my contributions under its terms.

> [!NOTE]
> Deleting the CLA section will lead to immediate closure of your PR and it will not be merged in.

davecrab committed:
Add adjustable text size setting to interface

Introduces a user-configurable text size (scale) setting, accessible via a slider in the interface settings. Updates CSS and Sidebar chat item components to respect the new --app-text-scale variable, and persists the setting in the store. Adds related i18n strings and ensures the text scale is applied globally and clamped to allowed values.

davecrab committed:
Merge branch 'dev' into feat/text-size

davecrab committed:
Refactor text scale logic into utility module

Moved all text scale related constants and functions from components and stores into a new utility module (src/lib/utils/text-scale.ts). Updated imports and usage in Interface.svelte and index.ts to use the new module, improving code organization and reusability.

davecrab commented:
> Screenrecordings?


https://github.com/user-attachments/assets/4ce71be7-dd1c-4afe-8aec-ded98bbbfbd1

Yea that would have helped

tjbck commented on src/lib/components/layout/Sidebar/ChatItem.svelte:
```diff
@@ -348,7 +348,7 @@
 	{#if confirmEdit}
 		<div
 			id="sidebar-chat-item"
-			class=" w-full flex justify-between rounded-xl px-[11px] py-[6px] {id === $chatId ||
```
This cannot be merged as it currently stands, Any workarounds for this?

silentoplayz commented:
When I increase the text size to 1.5×, the toggles in the UI appear squished (left side). They look normal when I use the browser’s zoom instead (right side). Is this something that can be fixed?
<img width="2560" height="1007" alt="image" src="https://github.com/user-attachments/assets/9f96573f-b16a-42bc-b34c-4897b43be5c3" />

davecrab committed:
Adjust sidebar chat scaling without extra classes

keep sidebar markup using existing Tailwind utility classes so chat items render identically pre-feature
move all text-scale sizing into app.css under the #sidebar-chat-item selectors
change the root font-size multiplier to use 1rem instead of an explicit 16px so browser/user preferences propagate

davecrab commented on src/app.css:
```diff
@@ -30,8 +30,21 @@
+
 html {
 	word-break: break-word;
+	font-size: calc(16px * var(--app-text-scale, 1));
```
Switched to 1 rem

davecrab commented:
> qq: I suppose this approach is superior because you can modify the text size in mobile settings as well?

Yea this is largely for mobile

davecrab committed:
Update Switch.svelte

Adjust toggles from fixed pixel to rem to scale with the text size

davecrab committed:
Update Interface.svelte

Updated label from 'Text Scale' to 'UI Scale'.
Added padding around slider

davecrab committed:
Update app.css

Added comments

tjbck commented:
Great addition, Thanks!
